### ISSUE TITLE
blob in preview

### ISSUE THREAD

**Issue by AlexMeier99** [mozilla/thimble.mozilla.org#1518]
When I open the preview of my thimble project there is always this blob on the screen...
![blob](https://cloud.githubusercontent.com/assets/17048656/17020118/31139400-4f40-11e6-822d-465bf79ca3f2.jpg)

So I inspected it in Firefox Inspector and got that:
![blob_inspected](https://cloud.githubusercontent.com/assets/17048656/17020133/47cf4bd0-4f40-11e6-8d95-f100cd458f0b.jpg)

I realized that it has to be related to the editor, so why is it not hidden?

**Comment by gideonthomas** [comment]
Good catch! This gives us a lead into where in the Brackets code to look for this.

**Comment by khanh2003** [comment]
yeah!
fucking blob!
i thought it was my screen's problem, but thank you for creating this issue.

**Comment by AlexMeier99** [comment]
Did you fixed?

**Comment by flukeout** [comment]
This is coming from the file tray, we could... 

* Add some CSS to hide it explicitly, or...
* Close the file tray when the full screen preview is opened

**Comment by Pomax** [comment]
Where is this actually coming from though? Can we find a way for it to not even show up in the first place rather than trying to hide it?

**Comment by humphd** [comment]
It's an artifact from Brackets, I think to do with selection of a file in the filetree.  We hide a bunch of the filetree (they have 2 sections for all files vs. open files), and I bet we missed something in there.

**Comment by AlexMeier99** [comment]
Today i tried the thimble preview again, triggerd by the github notification of the changes here.
So i don't see the blob any more, is it fixed?
(latest Firefox on Windows 7)

**Comment by humphd** [comment]
@AlexMeier99 it's an odd bug, and I don't think it happens all the time--which is partly why it hasn't been fixed in the past.  Part of what we need is clear steps to reproduce it.

**Comment by flukeout** [comment]
@AlexMeier99 for me it's always visible when the preview is in fullscreen mode. Is that where you were seeing it originally? To go full-screen, hit the icon on the far right of the preview bar...

![image](https://cloud.githubusercontent.com/assets/25212/22414193/777375ae-e672-11e6-8ee1-5f5605ec978f.png)

**Comment by AlexMeier99** [comment]
@flukeout So this is what is see in my thimble fullscreen preview:
![fullpage-screenshot-thimble](https://cloud.githubusercontent.com/assets/17048656/22422008/7c7f99a6-e6ea-11e6-9b76-f9f95e2f5e21.jpg)

**Comment by AlexMeier99** [comment]
No blob anymore?

**Comment by flukeout** [comment]
Before going fullscreen, select one of the files...

![image](https://cloud.githubusercontent.com/assets/25212/22432047/813aa37c-e6c8-11e6-8798-af09ffb53879.png)

That bring up the blob for me.

**Comment by AlexMeier99** [comment]
Ok now I got the same bug.
Then i checked the thimble code for the "fullscreen preview functions" called _enableFullscreenPreview_ and _disableFullscreenPreview_ (contained in:  https://raw.githubusercontent.com/mozilla/thimble.mozilla.org/fa3dde088dc016a562520e62a9df1f5558ba8f70/public/editor/scripts/editor/js/fc/bramble-ui-bridge.js  from line 282 to 294). I think these are called to show/hide the fullscreen preview.
Maybe we could remove the blob by adding this command to both functions:
$('#project-files-container').toggle();

**Comment by flukeout** [comment]
One thing I've noticed is that that little element gets a ``z-index: 18`` rule somewhere in the CSS. This places it above the full-screen preview. If we are able to get rid of that rule, that would solve this as well and might be an easier approach.

**Comment by flukeout** [comment]
Also, check out #1681 - it seems related, maybe we can just adjust the fullscreen preview z-index to be higher than the stuff that is currently overlapping it?

**Comment by flukeout** [comment]
Are you still working on this @AlexMeier99 - have you tried the Z-index solution?

**Comment by AlexMeier99** [comment]
Sorry i didn't i was quite busy during the last week.
But when checked #1681 i realized that @Mordax offered to solve this problem, so i stopped dealing with this issue a that moment.
Thanks in advance,
Alex

**Comment by flukeout** [comment]
Okay, Alex - thank you. I'm going to remove the Good First Bug label. If the work in #1618 fixes this, we can close this issue, otherwise I'll re-open it.

**Comment by AlexMeier99** [comment]
Ok thanks.

**Comment by humphd** [comment]
I can't reproduce this on staging (select file, go full screen or select file, drag file tree narrower).  Can someone else try and see if it still exists now that I've merged with upstream fixes:

https://bramble.mofostaging.net

**Comment by Mordax** [comment]
Using that URL in Chrome, I can recreate the image overlay issues in [1681](https://github.com/mozilla/thimble.mozilla.org/issues/1681) and the drag file tree narrower. The black blob however doesn't seem to be popping up. Running Thimble locally has the blob popping up most of the time.
